pkg/types/clusterconfig/clusterconfig.go

```diff
@@ -614,7 +545,7 @@ func AutoGenerateSpotConfig(awsClient *aws.Client, spotConfig *SpotConfig, regio
 	return nil
 }
 
-func (cc *Config) AutoFillSpot(awsClient *aws.Client) error {
+func (cc *Config) FillEmptyFields(awsClient *aws.Client) error {
 	if cc.SpotConfig == nil {
 		cc.SpotConfig = &SpotConfig{}
 	}
```

Comment:
Since this is a thin wrapper around `AutoGenerateSpotConfig()`, do you think it makes sense to move the contents of `AutoGenerateSpotConfig()` into here and remove that function? Or do we rely on this different access pattern elsewhere?

Reply:
Thing is, `AutoGenerateSpotConfig` is called in 2 places: in this function `FillEmptyFields` and in https://github.com/cortexlabs/cortex/blob/master/cli/cmd/lib_cluster_config.go#L416. Notice how the function in `lib_cluster_config_go` needs a `SpotConfig` whereas `FillEmptyFields` requires a `Config`. `FillEmptyFields` is also called in 2 places.

That being said, it's still a thin wrapper. You could move the contents of `AutoGenerateSpotConfig` inside `FillEmptyFields`, but it would be more appropriate to implement for `Config.SpotConfig`)

Reply:
Ah I see. In that case I think it's ok to leave it as is for now.